Feature/cafeteria 1

Seokyoung-Hong opened the pull request:
# 학식 보기 및 학식 등록 기능 추가
## 요약:
이 PR에서는 학식 보기와 학식 등록 기능을 포함한 주요 변경 사항을 추가합니다. 이 기능들은 사용자가 학식 정보를 조회하고 새로운 메뉴를 등록할 수 있도록 지원합니다.

## 주요 변경 사항:
1. 학식 등록 기능
    - `/meal/register/<meal_type>`: 주어진 식사 유형(중식 또는 석식)에 대해 메뉴를 등록하는 API
    - `/meal/register/delete/<meal_type>`: 등록된 메뉴를 삭제하기 위한 API
    - `/meal/register/delete_all`: 모든 메뉴를 삭제하는 API
    - `/meal/register/delete_menu`: 특정 메뉴를 삭제하는 API
    - `/meal/submit`: 임시 저장된 식단 정보를 확정 등록하는 API

2. 학식 보기 기능
    - `/meal/view`: 학식 정보를 `Carousel TextCard` 형태로 반환하는 API

3. Restaurant 클래스 추가
    - 식당 정보를 관리하고 메뉴를 추가, 삭제, 저장, 로드하는 기능을 포함한 클래스
    - `by_dict`, `by_id` 클래스 메서드를 통해 JSON 데이터 및 식당 ID를 기반으로 객체 생성
    - `add_menu`, `delete_menu`, `clear_menu` 메서드를 통해 메뉴 관리
    - `save_temp_menu`, `load_temp_menu` 메서드를 통해 임시 메뉴를 JSON 파일에 저장 및 로드
    - `submit` 메서드를 통해 임시 메뉴를 확정하여 저장


## 상세 설명:
  - `Restaurant` 클래스는 각 식당의 정보를 객체로 관리하며, 메뉴 등록 및 삭제, 임시 메뉴 저장 및 로드를 지원합니다.
  - `get_meals` 함수는 JSON 파일에서 식당 목록을 불러와 Restaurant 객체 리스트로 반환합니다.
  - `make_meal_cards`와 `meal_response_maker` 등의 유틸리티 함수를 통해 응답 형식을 일관되게 유지하고 있습니다.
  - 오류 처리를 위해 `handle_errors` 데코레이터를 사용하였습니다.
  - `view`나 `submit` api가 호출될 경우 자동 크롤링을 진행하도록 하기 위해 `check_tip_and_e` 데코레이터를 사용했습니다.
  
  
  ## 코드리뷰 포인트:
  - 코드가 Google Style에 맞추어 올바르게 작성되었는지 확인 부탁드립니다.
  - 코드의 가독성과 유지보수성을 고려한 구조인지 확인 부탁드립니다.
  - 에러가 발생할 수 있는 반례상황이 있는지 확인 부탁드립니다.
  - 오류 처리 로직이 적절히 구현되었는지 검토 부탁드립니다.
  - 파일처리 로직이 서버에 적용 가능한지 확인 부탁드립니다.

※ sandol/api_server/kakao에 있는 라이브러리는 저희 프로젝트 내용이 아닙니다. 무시 부탁드립니다.

Seokyoung-Hong committed:
Sync with main (#10)

* create: 초기 구조 생성

* Delete sandol/crawler/app.py

* create: 초기 구조 생성

---------

Co-authored-by: Cycrypto <jh01love00@gmail.com>
Co-authored-by: Cycrypto <jh01love@naver.com>

IvyKi committed:
Add files via upload

Feat: "추가 app.py, test.json"

body: app.py 내에 read_json(), get_meals() 함수 추가했습니다. 데이터를 가져오는 과정이 구현이 안돼서 임시로 test.json 파일을 만들어서 구현했습니다.

Seokyoung-Hong committed:
Crawler 폴더 이동

Sandol 폴더 밑으로 이동했습니다.

Seokyoung-Hong committed:
📰feat: 텍스트 카드 함수 생성

utils.py 파일 생성 및 TextCard 형태를 반환하는 함수 작성

Seokyoung-Hong committed:
📦feat: Update __init__.py

app.py에서의 사용을 위해 패키지들의 __init__.py 파일을 업데이트 합니다.

Seokyoung-Hong committed:
✨feat: 학식 발화 API 추가

식단 정보를 Carousel TextCard 형태로 반환합니다.

Seokyoung-Hong committed:
🙈chore: .gitignore 수정

- 중복 제거
- 무시 내용에 pycharm용 .idea 폴더 추가

Seokyoung-Hong committed:
🔨refactor: make_textcard 함수명 변경

PEP 8에 맞게 make_TextCard 함수명을 make_textcard로 변경

Seokyoung-Hong committed:
♻️ refactor: utils 임포트 상대경로로 수정

__init__.py의 utils 호출을 상대경로로 수정

Seokyoung-Hong committed:
♻️ refactor: Type Hinting for Mypy

정적 타입 체크기능 Mypy용 타입 힌트 추가

IvyKi committed:
:recycle: refactor: get_meals() 함수 변경

get_meals()의 인자 비워두고, read_json()파일 삭제했습니다.

IvyKi committed:
chore: cralwer file 위치 변경

sandol/cralwer 위치변경

IvyKi committed:
Delete crawler directory

crawler file path 수정

IvyKi committed:
chore: app.py 버전 수정

경로 수정하다가 파일이 이전 버전으로 올라갓군녀.. 버전 업데이트합니다

Seokyoung-Hong committed:
Update: kako-library

Seokyoung-Hong committed:
delete: 코드리뷰와 Merge를 위한 불필요 코드 제거

Seokyoung-Hong committed:
Feature/cafeteria 1 view (#20)

* add: add_quick_replies function

api_server의 utils.add_quick_replies 함수를 추가합니다. 이 함수는 카카오톡 응답 형식에 quickRelplies를 추가합니다.

* add: add_help_quick_reply

api_server의 utils.add_help_quick_reply를 반환하는 함수를 추가합니다. 도움말을 발화하는 QuickReply를 간단하게 추가하는 함수입니다.

* add: cafeteria_view.py, settings.py

id 조회, 식당 이름으로 식당 객체 불러오기

* fix: "identification" key 추가

* chore: by_id() 수정

식당 이름으로 조회해서 객체 생성하게 바꿨습니다

* fix: test.json path 수정

* add: kakao 라이브러리 적용

kakao 라이브러리를 임시로 로컬에 추가합니다.
이 라이브러리를 이용하여, 발화의 param 값을 추출하고, 각종 반환 값을 생성할 수 있습니다.

* delete & add: api_server/utils.py functions

api_server/utils.py의 함수를 수정합니다.
Kakao 라이브러리 추가로 불필요해진 함수들을 삭제합니다.
meal/view 처리를 위해 필요한 make_meal_cards 함수를 추가합니다.

* add: 재사용 가능한 컴포넌트 분리를 위한 api_server/settings.py 추가

재사용성이 높고, 변하지 않는 data를 저장하는 파일을 만들었습니다.
도움말 QuickReply와 TIP와 E동 식당 웹페이지 TextCard 가 추가됩니다.

* update: mea/view/ fuction

- kakao 라이브러리를 적용하고, 이를 이용하여 최적화 하였습니다.
- 사용자의 발화를 처리하여 사용자가 선택한 식단만 반환하는 기능을 추가했습니다.
- 도움말 Quickreply를 추가하였습니다.
- 세로로 3개 이하의 박스만을 출력할 수 있는 kakao 스킬의 규칙에 따라 반환 형태를 수정하였습니다. (기존에는 교내,교외를 구분하였습니다. 현재는 단순히 점심 저녁으로 나뉘며 마지막 줄에 E동과 TIP 식단 정보가 담긴 TextCard가 전송됩니다.)

* update: Kakako Library

* docs: module docstring 추가

대상 파일: sandol/app.py, sandol/api_server/utils.py, sandol/api_server/settings.py

* add: 다른 식당 탐색 Qurickreplies 반환 추가

학식 발화를 입력하면 도움말 Qurickreply와 식당 이름들로 이루어진 Qurickreplies를 함께 반환합니다.

* fix: cafeteria_view.py

- by_id() 함수 id조회 후 식당 이름으로 된 객체로 반환
- app.py의 get_meals() 함수 이동
- __init__ 초기화 방식 수정

* delete: app.py

get_meals() 함수 -> cafeteria_view.py 파일로 이동 및
app.py 삭제

* fix: 파일 구조 변경에 따른 __init__.py 수정

* fix: __init__.py 수정, settings.py 에러 제거

* fix: view, registration 기능 분리에 따른 method 수정

* add: registration.py 

registration 기능

- add(단일 메뉴 추가), delete(단일 메뉴 제거), clear(전체 temp 제거), submit(확정) method

* add: registration.py 

registration 기능

- add(단일 메뉴 추가), delete(단일 메뉴 제거), clear(전체 temp 제거), submit(확정) method

* delete: cafeteria-registration.py 및 관련 코드 삭제

* fix: view 브랜치-crawler 버전 업데이트

- registration.py 파일 삭제

* Delete sandol/crawler/cafeteria_registration.py

* update: Kakao-library 업데이트

* update: Kakao-library 업데이트에 따른 코드 수정

* fix: kakao library update에 따른 코드 수정 누락 처리

* refactor: 하위 Flask 버전을 위한 decorator 수정
app.post를 app.route("/", methods=["POST"]) 로 수정했습니다.

---------

Co-authored-by: Ivy Ki <ginahye1@gmail.com>

Seokyoung-Hong committed:
Feature/cafeteria 1 registration (#19)

* add: cafeteria.py, settings module

Restaurant class 재구현 -> 미완

* add: file 내 "id" key 추가

* Delete sandol/crawler/crawler directory

* fix: test.json, settings.py

* delete: sandol/crawler/app.py

app.py의 get_meals() 함수를 cafeteria_view.py로 이동, 동시에 app.py 불필요, delete

* add: Kakao 폴더 업로드

cafeteria-view에 적용된 이후 업데이트가 이루어진 코드로써 구조적 변화가 생겨 사용법이 바뀌었습니다.

* Add: 테스트용 식당 추가

테스트를 위해 산돌식당을 추가합니다.

* Add: cafeteria_registration.py 추가

식당의 메뉴 등록을 진행 하는 Restaurant 클래스가 포함된 파일을 추가합니다. 이를 initializing하는 관련 코드도 추가합니다.

* feat: 애플리케이션 파일 수정 및 식단 정보 등록 API 추가

* feature: temp_menu.json 파일 생성 용 method 추가

save_temp_menu() , load_temp_menu() 추가 및 submit() 수정
- submit 반환값 삭제

* chore: submit() method 수정(temp_menu.json 파일 영구삭제)

- self.temp_menu 초기화 및 submit 호출 시 temp_menu.json 영구 삭제
- add_menu() 내 중복 메뉴 등록 x 조건문 추가

* fix: registration.py method 수정

- save_temp_menu(), load_temp_menu() : 수정
- submit() : file.seek(0) 삭제, self.lunch 직접 access 삭제

* fix: 에러 메세지 호출 수정

print -> raise ValueError() 수정

* Update: Kakao library 업데이트

* add: testserver 동작을 위한 main.py 파일 추가

* fix: submit() 수정 및 자잘구리 fix

- submit() : lunch, dinner 개별 변화 시 개별 저장 -> submit_upload_menu() method 이용(복잡도 완화 목적)
- import settings -> from . import settings 수정
- add_menu(), dinner_menu() fix

* delete: unnecessary function make_textcard

Kakao Library 도입으로 불필요해진 함수를 제거합니다.

* feat: Add split_string function for string splitting

문자열을 구분자를 기준으로 분리하여 리스트로 반환하는 함수입니다. 학식 등록시 문자열로 입력받은 메뉴 문자열을 구분하기 위해 사용합니다.

* feat: make_meal_card and make_meal_cards function

식당의 식단 정보를 TextCard 형식으로 반환하는 make_meal_card와 식단 목록에 대해 식단 정보를 Carousel 형식으로 반환하는 함수입니다.

* feat: error_message function

에러를 받아 사용자에게 전달하기 위한 추가 메시지를 덧붙여 TextCard로 반환하는 함수입니다.

* feat: meal_registration api server code
식단 정보 등록을 위한 Flask 코드 전체를 update 합니다. meal_register는 임시로 식단 정보를 저장하고, meal_submit은 임시 저장된 식단 정보를 등록합니다.

* fix: self.temp_menu 인스턴스 삭제

- self.temp_menu 인스턴스 삭제에 따른 메서드 수정

* fix: import settings 상대경로 수정

* fix: utils 수정에 따른 코드 삭제 누락 해결

* feat: api_server/utils meal_response_maker function
식단 정보 미리보기 응답을 생성하는 함수입니다.
식단 등록, 식단 삭제에서의 코드 중복 문제를 해결합니다.

* feat: 식단 메뉴 삭제 기능 추가
식단 메뉴 삭제 기능을 추가합니다.
삭제하려는 식단 종류(점심, 저녁)의 메뉴들을 퀵리플라이로 반환하고, 이를 선택하면 메뉴가 삭제되는 함수입니다.

* Docs: add docstring

* update: kakao-library 코드리뷰 및 Merge를 위한 업데이트(코드 동기화)

* feat: 전체 메뉴 삭제 기능을 추가합니다.

* refactor: Refactor app.py

* refactor: 일관성 있는 queto 적용, module docstring 추가

* feat: handle_errors 데코레이터 추가
공통 오류 처리 데코레이터를 통해, 각 api 엔드포인트 함수에서 발생하는 별도로 Handling 되지 않는 모든 에러를 처리함으로써 5XX 에러를 원천적으로 차단합니다.

* refactor: 하위 Flask 버전 지원을 위한 수정
app.post 대신 app.route("/", methods=["POST"])를 사용합니다.

* feature: TIP, E동 레스토랑 식단 정보 등록

* chore: 모듈로 사용하기 위한, bookDownloader 수정
error raise: 이제 각 코드 도중 문제가 발생하면 오류 출력 대신 에러를 raise 합니다.
timeout 추가: 무제한 로딩 방지를 위해 requests.post 코드에 timeout 10초를 추가했습니다.

* fix: tip 가가식당, E동 레스토랑 identification code 할당

* fix: Restaurant 객체에 time, price 인스턴스 추가

registration_time, opening_time, price_per_person 인스턴스 추가

---------

Co-authored-by: Ivy Ki <144656120+IvyKi@users.noreply.github.com>

Seokyoung-Hong committed:
fix: 병합 실수 해결

Seokyoung-Hong committed:
fix: 병합실수 해결 crawler/settings.py

Seokyoung-Hong committed:
fix: handle_errors 함수 오류 수정

Seokyoung-Hong committed:
chore: 도움말 quickReply 제거

불필요한 quickReply를 제거합니다.

Seokyoung-Hong committed:
update: view 함수 수정

tip 지하|E동 식당 정보를 Restaurant 객체로 처리하도록 수정했습니다.
식단 정보를 등록시간 기준으로 정렬하도록 처리하는 코드를 주석으로 추가했습니다.

IvyKi committed:
fix: opening_time 인스턴스 -> datetime 객체 형태로 출력 수정

IvyKi committed:
feature: registration.py, view.py 합치기

- restaurant.py 하나에 registration.py, view.py 합치기, 인스턴스 통합
- __init__.py import Restaurant 호출 수정
- 자잘구리 settings, test.json 수정

IvyKi committed:
fix: registration_time 시간정보 추출 point 변경

+ 파일 이름이 restaurant 인 것보다 cafeteria.py가 더 적합한 것 같아 이름만 바꿨습니다

IvyKi committed:
Delete sandol/crawler/cafeteria_registration.py

브랜치 병합
cafeteria.py 파일로 정리

IvyKi committed:
Delete sandol/crawler/cafeteria_view.py

브랜치 병합
cafeteria.py로 파일 합침

IvyKi committed:
Delete sandol/crawler/restaurant.py

IvyKi committed:
fix: registration_time 지정 시 datetime 객체로 지정

IvyKi committed:
fix: by_dict() classmethod 생성 -> get_meals() 함수 수정

Seokyoung-Hong committed:
update: crawler의 각종 기능 지원을 위한 requirements.txt 수정

openpyxl, pandas, types-requests 추가

Seokyoung-Hong committed:
update: kakao 라이브러리 업데이트

Seokyoung-Hong committed:
feat: utils.py 각종 기능 추가 및 make_meal_card 관련 함수 수정

make_meal_card: 식단 등록 시간 표기 기능 추가 및 ItemCardComponent 반환 코드 주석으로 포함

error_message: BaseException의 경우 에러를 더 자세하게 출력하도록 수정

check_tip_and_e: "TIP 가가식당과 E동 레스토랑 정보를 업데이트하는 데코레이터

Seokyoung-Hong committed:
feat: api_server/settings.py 식단 등록 반환용 quick_reply 이관

식단 등록작업 중에 사용할 quick_reply들을 settings.py로 옮깁니다.

Seokyoung-Hong committed:
feat: utils.py meal_error_response_maker 함수 추가

식단 정보 에러 메시지를 반환하는 응답을 생성합니다.
현재 식단 개수 초과 시에 사용합니다.

Seokyoung-Hong committed:
refactor: 불필요 코드 삭제

Seokyoung-Hong committed:
feat: 등록 검증 API 추가와 에러 처리 및 검증 관련 각종 코드 수정

meal_view: 식단 등록 시간 순으로 정렬하도록 주석 해제 및 수정

meal_register: 메뉴 개수가 5개를 초과하는 경우 오류 문구 반환

meal_menu_delete: meal_error_response_maker함수 적

validation_menu: 메뉴 유효성 검사 API로 배열로 입력된 메뉴가 5개를 초과하는 경우 검증 오류 반환

Seokyoung-Hong committed:
chore: menu 개수 제한 코드 주석처리

TextCard 형식의 반환으로 코드가 작성되어 있음에 따라 불필요 코드 주석처리.

Seokyoung-Hong committed:
refactor: module 호출 부분 line to long 해결

Seokyoung-Hong committed:
Merge branch 'main' into feature/cafeteria-1

Seokyoung-Hong commented on sandol/crawler/cafeteria.py:
```diff
@@ -0,0 +1,273 @@
+import os
```
모듈 docstring을 작성해 주세요. 간단하게 해당 모듈(파일)이 어떤 기능을 담고 있는지만 작성하면 됩니다.

Seokyoung-Hong commented on sandol/crawler/cafeteria.py:
```diff
@@ -0,0 +1,273 @@
+from . import settings
+
+
+class Restaurant:
```
클래스 docstring을 작성해 주세요. 우리는 google style을 따르기로 했으므로, 아래 첨부된 링크를 참조하면 좋을 것 같습니다.
[docstring ](https://yosseulsin-job.github.io/Google-Python-Style-Guide-kor/#s3.8)

Seokyoung-Hong commented on sandol/crawler/cafeteria.py:
```diff
@@ -0,0 +1,273 @@
+    def submit_update_menu(self, menu_type):
+        """
+            submit(self)에서 사용. (복잡도 완화 목적)
+            :param menu_type: 업데이트할 메뉴 시간정보 ("lunch" 또는 "dinner").
```
우리는 google style을 따르기로 했으므로, docstring을 google style로 변경해주세요.
[docstring ](https://yosseulsin-job.github.io/Google-Python-Style-Guide-kor/#s3.8)

Seokyoung-Hong commented on sandol/crawler/__init__.py:
```diff
@@ -0,0 +1,30 @@
+
+# view 모듈에서 사용자가 직접 호출할 수 있는 함수 정의.
+# 각 점포 별 계정 id로 식별
+def get_view(identification):
```
함수 위 주석 대신 docstring을 활용해 주세요

Seokyoung-Hong commented on sandol/crawler/__init__.py:
```diff
@@ -0,0 +1,30 @@
+    return Restaurant.by_id(identification)
+
+
+# registration 모듈에서 사용자가 직접 호출할 수 있는 함수 정의
```
불필요한 주석이므로 삭제해주세요.

Seokyoung-Hong commented on sandol/crawler/cafeteria.py:
```diff
@@ -0,0 +1,273 @@
+
+        if restaurant_name:
+            current_dir = os.path.dirname(__file__)
+            filename = os.path.join(current_dir, 'test.json')
```
한 파일 안에서, Queto(따옴표)는 일관적으로 사용해주세요.
' 만 사용하거나, "만 사용해주세요.

결정한 결과에 맞추어 파일 전체를 수정해야합니다!

Seokyoung-Hong commented on sandol/crawler/cafeteria.py:
```diff
@@ -0,0 +1,273 @@
+                start = dt.datetime.strptime(start_time, '%p %I:%M').time()
+                end = dt.datetime.strptime(end_time, '%I:%M').time()
+
+                self.opening_time.append([start.strftime("%p %I:%M"), end.strftime("%I:%M")])
```
코드가 80자를 초과해요. append 내부를 줄바꿈 하는 것이 좋을 것 같아요.

Seokyoung-Hong commented on sandol/crawler/cafeteria.py:
```diff
@@ -0,0 +1,273 @@
+                data = json.load(file)
+        except json.decoder.JSONDecodeError:
+            data = []
+        except FileNotFoundError:
```
이 경우에는,
```python
try:
    # Some code that may raise FileNotFoundError
except FileNotFoundError as exc:
    raise FileNotFoundError(f'{filename} 파일을 찾을 수 없습니다.') from exc
```

이런 식으로 
as exc와 from exc로 에러를 연장처리 하는 것이 좋습니다. (동일한 종류의 에러를 만드니까요!)

Seokyoung-Hong commented on sandol/crawler/cafeteria.py:
```diff
@@ -0,0 +1,273 @@
+            raise ValueError(f"레스토랑 '{self.name}'가 test.json 파일에 존재하지 않습니다.")
+
+        with open(filename, 'w', encoding='utf-8') as file:
+            json.dump(data, file, ensure_ascii=False, indent=4)                     # json data 한꺼번에 test.json으로 덮어씌우기
```
## 주석 작성 규칙

1. 주석은 코드 옆 2칸 뒤에 작성해주세요.
2. 주석이 코드와 연결되기 위해 고의적으로 떨어져야 하는 경우, 주석을 코드 옆에 작성하는 것이 아니라, 서로 연결되는 주석 내용을 한 번에 작성하는 것이 좋습니다.
3. 주석을 포함한 코드가 80자를 넘기지 않게 유의해야합니다!

### 예시 코드 및 주석

```python
for restaurant_data in data:
    if restaurant_data["name"] == self.name:  # 식당 검색
        restaurant_data["lunch_menu"] = self.submit_update_menu("lunch")  # 점심 메뉴 변경 사항 존재 시 submit
        restaurant_data["dinner_menu"] = self.submit_update_menu("dinner")  # 저녁 메뉴 변경 사항 존재 시 submit
        restaurant_data["registration_time"] = dt.datetime.now().isoformat()  # registration time update
        restaurant_data["opening_time"] = self.opening_time  # opening time update
        restaurant_data["price_per_person"] = self.price_per_person  # 가격 update
        restaurant_found = True
        break

if not restaurant_found:
    raise ValueError(f"레스토랑 '{self.name}'가 test.json 파일에 존재하지 않습니다.")

with open(filename, 'w', encoding='utf-8') as file:
    json.dump(data, file, ensure_ascii=False, indent=4)  # json data 한꺼번에 test.json으로 덮어씌우기
```

## 개선된 주석 작성 예시
주석이 코드와 연결되기 위해 고의적으로 떨어져야 하는 경우, 다음과 같이 주석을 한 번에 작성하는 것이 좋습니다:

```python
for restaurant_data in data:
    if restaurant_data["name"] == self.name:  # 식당 검색
        # 점심 및 저녁 메뉴 변경 사항 존재 시 submit
        restaurant_data["lunch_menu"] = self.submit_update_menu("lunch")
        restaurant_data["dinner_menu"] = self.submit_update_menu("dinner")

        # registration time, opening time, 가격 update
        restaurant_data["registration_time"] = dt.datetime.now().isoformat()
        restaurant_data["opening_time"] = self.opening_time
        restaurant_data["price_per_person"] = self.price_per_person
        
        restaurant_found = True
        break

if not restaurant_found:
    raise ValueError(f"레스토랑 '{self.name}'가 test.json 파일에 존재하지 않습니다.")

with open(filename, 'w', encoding='utf-8') as file:
    json.dump(data, file, ensure_ascii=False, indent=4)  # json data 한꺼번에 test.json으로 덮어씌우기
```
이렇게 하면 코드와 주석이 더 명확하게 구분되어 가독성이 향상됩니다.

Seokyoung-Hong commented on sandol/crawler/cafeteria.py:
```diff
@@ -0,0 +1,273 @@
+        if os.path.exists(temp_menu_path):
+            os.remove(temp_menu_path)
+        else:
+            raise ValueError("temp_menu.json file doesn't exist")
```
파일 삭제가 목적인 코드에서 파일이 존재하지 않음에 대한 에러는 발생할 필요가 없을 것 같습니다.

Seokyoung-Hong commented on sandol/crawler/cafeteria.py:
```diff
@@ -0,0 +1,273 @@
+        """
+            print() test 시 출력 메세지 가시성 완화 목적
+        """
+        return (f"Restaurant: {self.name}, Lunch: {self.lunch}, Dinner: {self.dinner}, "
```
코드가 너무 깁니다. 코드 길이를 줄여주세요. 
 `(` 뒤에 값을 넣지 말고 바로 줄바꿈 할 수 있습니다.
```python
temp_data = ["data",
             "data2"]
```
대신,
```python
temp_data = [
    "data",
    "data2"
    ]
```
이렇게 할 수 있습니다.
물론 위의 방법도 틀린 방법은 아닙니다.

Seokyoung-Hong commented on sandol/crawler/cafeteria.py:
```diff
@@ -0,0 +1,273 @@
+            raise ValueError("meal_time should be 'lunch' or 'dinner'.")
+
+        # save temp_menu.json
+        self.save_temp_menu()
```
add_menu와 delete_menu 메서드의 코드가 거의 유사합니다. 이를 하나의 메서드로 통합하고, 추가나 삭제를 인자로 받아 처리하면 중복을 줄일 수 있습니다.

Seokyoung-Hong commented on sandol/crawler/cafeteria.py:
```diff
@@ -0,0 +1,273 @@
+        class_name = f"{data['name']}"
+        new_class = type(class_name, (Restaurant,), {})
+
+        return new_class(data["name"], data["lunch_menu"], data["dinner_menu"], data["location"], registration_time)
```
1. return 코드가 너무 길어요. 한 줄에 80자 이하로 작성해야 해요. 괄호를 씌워 줄바꿈을 하는 것이 좋을것 같아요.
2. new_class 대신, cls를 사용하는 것이 좋을 것 같아요.

Seokyoung-Hong commented on sandol/crawler/cafeteria.py:
```diff
@@ -0,0 +1,273 @@
+            "산돌식당": ["오후 12:00-1:00 / 오후 5:30-6:30", 6500],
+            "TIP 가가식당": ["오전 11:00-2:00 / 오후 5:00-6:50", 6000],
+            "E동 레스토랑": ["오전 11:30-1:50 / 오후 4:50-6:40", 6500]
+        }
```
rest_info 메서드에서 레스토랑 정보가 하드코딩되어 있습니다. 이는 유지보수 시 불편할 수 있습니다. 이러한 데이터를 외부 JSON 파일이나 데이터베이스에서 불러오는 것이 좋습니다.
아니면, settings.py와 같은 상수 전용 관리 모듈을 이용하는 것이 좋습니다.

Seokyoung-Hong commented on sandol/crawler/cafeteria.py:
```diff
@@ -0,0 +1,273 @@
+        }
+
+        if self.name in info:
+            time_info, self.price_per_person = info[self.name]
+            time_slots = time_info.split(" / ")
```
type hint를 추가하는 것이 좋을 것 같아요!
이를 통해 `time_info.split(" / ")` 코드에서 linter의 도움을 받을 수 있고,
코드를 검증하기가 쉬워져요.

Seokyoung-Hong commented on sandol/crawler/cafeteria.py:
```diff
@@ -0,0 +1,273 @@
+                restaurant_data["opening_time"] = self.opening_time                 # opining time update
+                restaurant_data["price_per_person"] = self.price_per_person         # 가격 update
+                restaurant_found = True
+                break
```
이 if문의 각 라인에 대한 주석을 일반화하여 달 수 있을 것 같아요.
뿐만 아니라, 주석을 포함한 코드가 80자를 초과해서는 안돼요!
만약 주석으로 인해 80자를 초과한다면, 주석을 코드 위로 올릴 수 있어요.

Seokyoung-Hong commented on sandol/crawler/cafeteria.py:
```diff
@@ -0,0 +1,273 @@
+            filename = os.path.join(current_dir, 'test.json')
+
+            with open(filename, 'r', encoding='utf-8') as file:
+                data = json.load(file)
```
이 코드가 클래스 전체에서 반복되는 것을 확인했어요.
GPT는 아래와 같이 static method를 추가해 처리하는 것을 추천해요!

```python
    @staticmethod
    def get_file_path(filename):
        """
        주어진 파일 이름에 대한 절대 경로를 반환하는 유틸리티 함수
        """
        current_dir = os.path.dirname(__file__)
        return os.path.join(current_dir, filename)

    @staticmethod
    def load_json_file(filename):
        """
        JSON 파일을 읽어오는 유틸리티 함수
        """
        filepath = Restaurant.get_file_path(filename)
        try:
            with open(filepath, 'r', encoding='utf-8') as file:
                return json.load(file)
        except (json.decoder.JSONDecodeError, FileNotFoundError):
            raise FileNotFoundError(f"{filename} 파일을 찾을 수 없습니다.")

    @staticmethod
    def save_json_file(data, filename):
        """
        데이터를 JSON 파일로 저장하는 유틸리티 함수
        """
        filepath = Restaurant.get_file_path(filename)
        with open(filepath, 'w', encoding='utf-8') as file:
            json.dump(data, file, ensure_ascii=False, indent=4)
```

Seokyoung-Hong commented on sandol/crawler/cafeteria.py:
```diff
@@ -0,0 +1,273 @@
+                self.opening_time.append([start.strftime("%p %I:%M"), end.strftime("%I:%M")])
+
+        else:
+            raise ValueError(f"레스토랑 '{self.name}'에 대한 정보를 찾을 수 없습니다.")
```
1. opening_time을 문자열로 저장하고 있는데, 이를 datetime객체로 저장하면 클래스를 더 잘 활용할 수 있을 것 같아요.
또한, `@property` decorator를 이용해 lunch_time_str, dinner_time_str 같은 메서드를 str으로 반환하도록 하면 더욱 더 유용할 수 있어요. (API 팀의 요구사항이 아닌 단순 리뷰 및 제안입니다.)
2. GPT는 시간 분석을 메서드 또는 함수로 분리하는 것을 추천해요. 이렇게 하면, 반환 형식이 어떻게 될지 더욱더 쉽게 이해할 수 있을것 같아요.
```python
def parse_time(self, time_str):
        """
        시간 문자열을 datetime.time 객체로 변환하는 메서드
        """
        time_str = time_str.strip().replace('시', ':').replace('분', '')
        if '오전' in time_str:
            time_str = time_str.replace('오전', 'AM ')
        elif '오후' in time_str:
            time_str = time_str.replace('오후', 'PM ')

        if ':' not in time_str.split()[1]:
            time_str += ':00'

        return dt.datetime.strptime(time_str, '%p %I:%M').time()
```

Seokyoung-Hong commented on sandol/crawler/ibookcrawler.py:
```diff
@@ -0,0 +1,188 @@
+        self.name = "E동 레스토랑"
+        self.opening_time = "오전 11:30-13:50 / 오후 4:50-18:40"
+        self.location = "E동 1층"
+        self.price_per_person = 6500
```
이 부분은 settings.py로 옮길 수 잇을 것 같아요.

Seokyoung-Hong committed:
refactor: rename bookDownloader to ibookdownloader and add module docstring

1. 파일명 수정: camel case 지양 규칙 적용
2. 모듈 docstring 추가

Seokyoung-Hong committed:
refactor: api_serer/utils.py 주석 깔끔하게

Seokyoung-Hong committed:
refactor: settings.py 등록용 퀵리플라이 목록 리스트로 전환

Seokyoung-Hong committed:
refactor: 주석을 줄였습니다.

Seokyoung-Hong committed:
updage: kakao library

Seokyoung-Hong committed:
fix: meal/view 비어있는 carousel 처리 추가

Seokyoung-Hong committed:
fix: 식단 정보가 없을 경우에 대한 예외처리 추가

Seokyoung-Hong committed:
update: fastAPI로 프레임워크 마이그레이션
1. FastAPI에 맞게 비동기 함수를 사용합니다.
2. 에러 처리를 FastApi에서 지원하는 exception_handler를 사용하도록 수정했습니다.
3. check_tip_and_e 데코레이터도 비동기 함수에 맞게 수정했습니다.

Seokyoung-Hong committed:
chore: 불필요한 내용 제거

Seokyoung-Hong committed:
delete: 불필요 파일 삭제

Seokyoung-Hong committed:
refactor: meal module 구조

조직화 및 유지보수를 개선하기 위해 meal 모듈 구조를 리팩토링합니다. meal API 엔드포인트 정의를 별도의 모듈로 이동하여 관심사의 분리를 강화합니다. 이를 통해 API 엔드포인트를 보다 쉽게 관리하고 업데이트할 수 있습니다.

- meal API 엔드포인트 정의를 별도의 모듈로 이동
- __init__.py 파일을 업데이트하여 meal API 모듈을 임포트
- __all__ 리스트를 업데이트하여 meal API 모듈을 포함
이 리팩토링은 API 서버의 전체 구조를 개선하고 유지보수 및 확장이 용이하도록 합니다.

Cycrypto commented on sandol/app.py:
```diff
@@ -1,11 +1,24 @@
 
-app = Flask(__name__)
+from .api_server.meal import meal_api
+from .api_server.utils import error_message
```
모듈 임포트에서 상대경로를 사용하셨는데, 이유가 있으실까요?
`sandol.api_server.meal`처럼 명확한 경로를 쓰는게 좋아보일거 같아서요..!

Seokyoung-Hong replied:
리뷰 감사합니다. 피드백을 반영하여 코드를 수정했습니다.
- 처음 코드를 작성할 때, app.py의 위치와 베포시 폴더 구조를 명확히 인지하지 못해서 구조의 수정을 염두하고 상대경로로 코드작성을 시작했습니다. 하지만 이제 코드 구조의 틀이 잡혔기 때문에 절대경로로 수정했습니다.

Cycrypto commented on sandol/api_server/meal.py:
```diff
@@ -0,0 +1,294 @@
+)
+from ..crawler import (
+    get_registration, Restaurant, get_meals
+)
```
여기도 마찬가지로 상대경로로 모듈을 import하고 있네요!

Seokyoung-Hong replied:
리뷰 감사합니다. 피드백을 반영하여 코드를 수정했습니다.

Cycrypto commented on sandol/api_server/meal.py:
```diff
@@ -0,0 +1,294 @@
+    restaurant.load_temp_menu()
+
+    # 카카오에서 전달받은 menu 파라미터를 구분자를 기준으로 분리해 리스트로 변환
+    assert payload.detail_params is not None
```
이부분이 이해가 잘 가지 않네요. 추가적인 설명 부탁드립니다.
들어올 값이 None이 아님이 보장된다의 의미로 사용하신거라면 굳이 존재 이유가 궁금해서요!

Cycrypto replied:
그게 아니라면 이에 대한 추가적인 예외처리가 필요하지 않나 싶습니다.

```python
    if payload.detail_params is None or "menu" not in payload.detail_params:
        return JSONResponse(
            meal_error_response_maker("메뉴 정보가 없습니다.").get_dict(),
            status_code=400
        )
```

Cycrypto commented on sandol/api_server/meal.py:
```diff
@@ -0,0 +1,294 @@
+
+
+@meal_api.post("/register/delete/{meal_type}")
+async def meal_delete(meal_type: str, request: Request):
```
meal_register, delete를 포함해 대부분 유사한 코드 흐름을 보이고 있습니다. 중복되는 코드가 많다는 의미인데, 따로 떼서 다른 모듈로 분리해보시는건 어떤가요? 예를들면 Payload 변환이나, 예외처리부분이라던가 그렇습니다.

Seokyoung-Hong replied:
리뷰 감사합니다. 피드백을 반영하여 코드를 수정했습니다.

이 부분을 발견하고 수정하려고 시도했으나 view와 register를 함께 통합해서 중복을 제거하려니 payload 변환을 제외하면 중복이 딱히 제거되는 것이 없어서 포기했습니다.
하지만 피드백 덕분에, register에서는 중복을 다수 줄일 수 있음을 확인하였고 수정했습니다.

- get_payload_and_restaurant 함수 추가, 해당함수는 API 로직과 밀접하게 연관되며, 해당 파일 외에서는 사용되지 않으며, utils.py의 다른 함수들과 다르게 FastAPI의 Request 객체 의존성이 있기 때문에 utils.py 대신, meal.py 함수에 작성하였습니다.

Cycrypto commented on sandol/api_server/utils.py:
```diff
@@ -0,0 +1,252 @@
+from functools import wraps
+import traceback
+
+from .settings import CAFETRIA_REGISTER_QUICK_REPLY_LIST
+from .kakao.response.components.card import ItemCardComponent
+from .kakao.response import KakaoResponse
+from .kakao.response.components import (
+    CarouselComponent, TextCardComponent, SimpleTextComponent)
+from ..crawler import Restaurant
+from ..crawler.ibookcrawler import BookTranslator
+from ..crawler.ibookdownloader import BookDownloader
```
여기도..

Seokyoung-Hong replied:
리뷰 감사합니다. 피드백을 반영하여 코드를 수정했습니다.

Cycrypto commented on sandol/api_server/utils.py:
```diff
@@ -0,0 +1,252 @@
+
+    # return response
+
+    description = ""
+    if not menu_list:
+        description = "식단 정보가 없습니다."
+    else:
+        for menu in menu_list:
+            description += f"{menu}\n"
```
```python
description = "\n".join(menu_list) if menu_list else "식단 정보가 없습니다."

```

Seokyoung-Hong replied:
리뷰 감사합니다. 피드백을 반영하여 코드를 수정했습니다.

Cycrypto commented on sandol/api_server/utils.py:
```diff
@@ -0,0 +1,252 @@
+    Returns:
+        KakaoResponse: 식단 정보 미리보기 응답
+    """
+    response = KakaoResponse()
+    simple_text = SimpleTextComponent("식단 정보 미리보기")
+
+    response = response + simple_text + lunch + dinner
+    for quick_reply in CAFETRIA_REGISTER_QUICK_REPLY_LIST:
+        response.add_quick_reply(quick_reply)
+    return response
```
```python
response = KakaoResponse() + SimpleTextComponent("식단 정보 미리보기") + lunch + dinner
response.add_quick_replies(CAFETRIA_REGISTER_QUICK_REPLY_LIST)
return response
```
예시입니다.
조금 더 간소화가 가능할 거 같아요.

Seokyoung-Hong replied:
리뷰 감사합니다. 피드백을 반영하여 코드를 수정했습니다.
하지만, add_quick_replies 메서드가 kakao 라이브러리에 존재하지 않아, 퀵리플라이에 대한 반복문은 수정되지 못했습니다.

Cycrypto commented on sandol/api_server/utils.py:
```diff
@@ -0,0 +1,252 @@
+    return TextCardComponent(title="오류 발생", description=message)
+
+
+def check_tip_and_e(func):
```
파일 경로와 수정시장 검증 로직을 함수로 분리하여 중복을 피할 수 있지 않을까 싶습니다

Cycrypto reviewed (commented):
1차 리뷰 (API)

Seokyoung-Hong committed:
refactor: utils.py/meal_response_maker 코드 간소화

반복적이고 단순한 연산을 하나로 통일하여 코드를 간소화 했습니다.

Seokyoung-Hong committed:
refactor: 모듈 import 절대 경로 사용
api_server/__init__.py를 제외한 코드에서의 import문을 상대경로에서 절대경로로 수정했습니다.

Seokyoung-Hong committed:
refactor: 불필요 코드 삭제

Seokyoung-Hong committed:
refactor: Payload와 Restaurant 객체를 반환하는 공통 함수 분리
- 요청에서 Payload 객체와 등록된 Restaurant 객체를 반환하는 공통 함수 get_payload_and_restaurant를 추가했습니다.
- meal_register 및 meal_delete 함수에서 중복된 코드를 제거하고, get_payload_and_restaurant 함수를 사용하도록 수정했습니다.
- 코드의 일관성과 재사용성을 높였습니다.

Seokyoung-Hong committed:
refactor: 주석 수정
- TODO 주석에 작성자를 추가했습니다.
- 잘못된 docstring 구조를 수정했습니다.

Seokyoung-Hong committed:
refactor: menu_list 처리를 더 간결하게 변경
- 기존의 반복문을 사용하여 description을 구성하던 코드를 간결한 단일 라인 표현으로 변경했습니다.
- menu_list가 비어 있을 경우 "식단 정보가 없습니다."를 반환하고, 그렇지 않으면 메뉴 항목들을 줄바꿈으로 연결하여 반환하도록 수정했습니다.
- 코드의 가독성과 효율성을 높였습니다.

Seokyoung-Hong committed:
refactor: 주석 수정 및 코드 간소화

IvyKi committed:
fix: tip메뉴 "*복수메뉴*" 문구 필터링

Seokyoung-Hong reviewed (commented):
코드리뷰

Seokyoung-Hong committed:
refactor: 식당 정보를 반환하는 API 엔드포인트 추가

식단 목록에 식당 정보를 확인하는 버튼을 추가했습니다.
식당 정보를 반환하는 API 엔드포인트를 추가했습니다. 이를 통해 클라이언트는 특정 식당의 정보를 요청할 수 있습니다. API 응답으로는 식당의 이름, 영업 시간, 위치, 가격 등의 정보가 포함됩니다. 또한, 식당 위치를 확인할 수 있는 네이버 지도 링크도 제공됩니다. 이를 통해 사용자는 편리하게 식당 정보를 확인할 수 있습니다.

Seokyoung-Hong committed:
update: S3 사용할 수 있도록 코드 수정

Seokyoung-Hong committed:
refactor: Update requirements.txt with necessary dependencies

Seokyoung-Hong committed:
refactor: Add Dockerfile for containerization

Seokyoung-Hong committed:
refactor: Update import statements for API modules

Seokyoung-Hong committed:
refactor: Update import statements for API modules

Seokyoung-Hong committed:
fix: 병합 도중 소실된 import 문과 fastAPI로의 프레임워크 수정에 따른 함수 위치 변경

Seokyoung-Hong committed:
fix: s3 파일처리 누락 부분 추가

Seokyoung-Hong committed:
fix: s3 적용 누락 수정

Seokyoung-Hong committed:
fix: 코드의 모든 datetime 객체를 KST로 시간대 수정
datetime을 이용하는 모든 코드가 시간대가 설정되지 않아, UTC 기반으로 설정되어 있어, 출력시 한국시간이 아닌 UTC 시간대로 출력되는 문제가 있어 해결했습니다.

Seokyoung-Hong committed:
update: check_tip_and_e 데코레이터 함수 알고리즘 수정
1. 새 파일로의 다운로드가 필요한지 판별합니다.
    - /tmp/data.xlsx 가 존재하지 않는 경우
    - data.xlsx 저장 시간이 지난주 수요일 이전인 경우(xlsx 파일 자체의 생성 시간을 가져오기 떄문에, 동일한 파일이 다시 다운되더라도 저장 시간은 변경되지 않습니다.)
2. 메뉴 업데이트가 필요한지 판별합니다.
    - tip 식당의 등록시간이 오늘 이전인 경우

Seokyoung-Hong committed:
add: 샘플용 data.xlsx 추가

IvyKi committed:
fix: 코드리뷰 반영 cafeteria.py 코드스타일 수정

IvyKi committed:
fix: 코드리뷰 반영 ibookcrawler.py 코드스타일 수정

Seokyoung-Hong committed:
fix: check_tip_and_e datetime 객체비교문제 수정
함수에서 사용하는 두 datetime 객체의 시간대를 KST로 변환하는 코드를 추가했습니다.

Seokyoung-Hong committed:
fix: 파일 위치 lamda 용으로 수정

Seokyoung-Hong committed:
fix: 파일 수정을 /tmp 밑에서 동작하도록 코드 수정
- 기존 업데이트에서 누락된 부분이 발견되어 수정합니다.

Seokyoung-Hong committed:
fix: ibookcrawler.py submit시 registration_time 정보를 함께 수정

Seokyoung-Hong committed:
update: cafeteria.by_id 메서드에 download 인자 추가
- download=True 인 경우 s3에서 파일 다운로드 안함
- mea submit에서 불필요하게 s3 download를 여러번 해 코드 실행시간이 증가하는걸 방지함

Seokyoung-Hong committed:
fix: Restaurant.delete_menu에 early return 추가

IvyKi committed:
fix: cafeteria.py docstring 수정

Seokyoung-Hong committed:
refactor: api_server/meal.py 의존성 주입 방식으로 변경

FastAPI가 지원하는 의존성 주입 방식을 이용해 Payload 객체를 더 쉽게 이용할 수 있습니다.
- 요청 데이터를 Payload 객체로 변환하는 parse_payload 함수 추가
- meal_delete 엔드포인트에서 Depends를 사용하여 Payload 객체 주입
- get_payload_and_restaurant 함수 삭제

Seokyoung-Hong committed:
fix: api_server/utils.py UnboundLocalError 문제 해결
- data.xlsx이 존재하지 않을 경우 tip 객체가 생성되지 않아 생기던 문제를 해결했습니다.

Seokyoung-Hong committed:
refactor: parse_payload 함수 이동
parse_payload 함수는 api_server에서 범용적으로 사용될 함수이기 때문에 utils.py로 이동합니다.

Seokyoung-Hong committed:
feat: replace internal Kakao library with official kakao-chatbot (#24)

내부에 임시로 포함되어 있던 Kakao 라이브러리를 정식 배포된 kakao-chatbot 라이브러리로 교체.

- `api_server/kakao` 폴더 삭제
- 기존 `from api_server.kakao import ~~` 구문을 `from kakao_chatbot import ~~`로 변경
- `requirements.txt`에 `kakao-chatbot` 추가

Cycrypto commented on sandol/api_server/meal.py:
```diff
@@ -0,0 +1,338 @@
+
+    # 퀵리플라이 추가
+    # 현재 선택된 식단을 제외한 다른 식당을 퀵리플라이로 추가
+    if target_cafeteria:
+        response.add_quick_reply(
+            label="모두 보기",
+            action="message",
+            message_text="테스트 학식",  # TODO(Seokyoung_Hong): 베포 시 '테스트' 제거
+        )
+    for rest in cafeteria_list:
+        if rest.name != target_cafeteria:
+            response.add_quick_reply(
+                label=rest.name,
+                action="message",
+                # TODO(Seokyoung_Hong): 베포 시 '테스트' 제거
+                message_text=f"테스트 학식 {rest.name}",
+            )
+
```
여기도 테스트가 제거된 상태인가요

Cycrypto commented on sandol/api_server/meal.py:
```diff
@@ -0,0 +1,338 @@
+    bf_standard: list[Restaurant] = []
+    for r in restaurants:
+        if r.registration_time.tzinfo is None:
+            print("등록시간에 시간대 정보가 없어 9시간을 더해 KST로 변환합니다.")
```
print를 대체할 수 있는 방법이 있을거 같아요

Cycrypto commented on sandol/api_server/meal.py:
```diff
@@ -0,0 +1,338 @@
+        response.add_quick_reply(
+            label="모두 보기",
+            action="message",
+            message_text="테스트 학식",  # TODO(Seokyoung_Hong): 베포 시 '테스트' 제거
```
제거가 되어야 하는 부분인가요

Cycrypto commented on sandol/api_server/meal.py:
```diff
@@ -0,0 +1,338 @@
+        label="메뉴 보기",
+        action="message",
+        # TODO(Seokyoung_Hong): 베포 시 '테스트' 제거
+        message_text=f"테스트 학식 {restaurant_name}"
```
여기도입니다

Cycrypto commented on sandol/api_server/settings.py:
```diff
@@ -0,0 +1,37 @@
+NAVER_MAP_URL_DICT = {
+    "미가식당": "https://naver.me/xEMZ6QdE",
+    "세미콘식당": "https://naver.me/xQ8Khcho",
+    "산돌식당": "https://naver.me/xEMZ6QdE",
```
이것도 제거가 되어야 하는부분인가요

Cycrypto commented on sandol/api_server/utils.py:
```diff
@@ -0,0 +1,312 @@
+    if isinstance(message, BaseException):
+        exception_type = type(message).__name__
+        exception_message = str(message)
+        exception_traceback = "".join(  # TODO(Seokyoung_Hong): 베포시 주석 처리
+            traceback.format_tb(message.__traceback__))
```
이부분도 처리가 안되어있는거 같아요

Cycrypto commented on sandol/api_server/utils.py:
```diff
@@ -0,0 +1,312 @@
+        detailed_message = (
+            f"예외 타입: {exception_type}\n"
+            f"예외 메시지: {exception_message}\n"
+            f"트레이스백:\n{exception_traceback}"  # TODO(Seokyoung_Hong): 베포시 주석 처리
```
동일합니다

Cycrypto commented on sandol/bucket/common.py:
```diff
@@ -0,0 +1,30 @@
+from botocore.exceptions import NoCredentialsError, PartialCredentialsError
+
+BUCKET_NAME = "sandol-bucket"
+FILE_KEY = "test.json"
```
파일 명 변경이 필요해보입니다. 혹시 권한이 없으시면 말씀해주세요

Cycrypto commented on sandol/crawler/test.json:
```diff
@@ -0,0 +1,120 @@
+[
```
아 test.json을 원래 s3에 있는 파일명으로 변경하셔야 할거 같네요..!

Cycrypto reviewed (changes requested):
확인했습니다

Seokyoung-Hong committed:
fix: 퀵리플라이 텍스트에서 'test' 제거

퀵리플라이의 'test 학식'을 '학식'으로 수정

Seokyoung-Hong committed:
fix: 주석 처리된 Traceback 메시지

디버깅을 위해 추가했던 오류시 Traceback 메시지를 베포를 위해 주석 처리했습니다.

Seokyoung-Hong committed:
diff
코드 복사
feat: logging 모듈을 이용해 로깅 기능 추가

- logging 모듈을 사용해 'sandol_logger'라는 이름의 로거를 추가했습니다.
- 주요 이벤트(서버 시작, 예외 처리, 엔드포인트 접근 등)에 대해 로그가 남도록 설정했습니다.

Seokyoung-Hong committed:
refactor: print문을 logger로 변경

- 기존의 print문을 전부 logger를 사용해 대체했습니다.
- 더 나은 로그 관리를 위해 정보성 메시지에는 logger.info, 오류 발생 시에는 logger.error를 사용했습니다.

Seokyoung-Hong commented:
현재 카카오톡 오픈빌더에서는 '학식'과 같은 의도된 발화 앞에 '테스트'라는 문구가 추가되도록 설정되어 있습니다. 이는 기존 '학식' 블록과 스킬을 변경하지 않고, 별도의 블록과 스킬로 개발하기 위해서입니다.

이에 맞춰 PR에 올린 코드에서도 '테스트 학식' 발화를 이용해 동작하도록 설정했습니다. 코드 리뷰 이후, 해당 코드를 배포 가능한 상태로 수정하였습니다.

다만, 카카오톡 오픈빌더 설정은 이번에 수정하지 않았으며, 실제 배포 시에는 오픈빌더 설정을 수정 및 배포해야 합니다.
